api/src/main/resources/api/service_openapi.yaml

```diff
@@ -92,6 +114,21 @@ components:
         boolVal:
           type: boolean
 
+    EntityDataset:
+      description: A selection of entity attributes and filters on them.
+      type: object
+      properties:
+        entityVariable:
+          description: The variable to use for the top level entity in the filter.
+          type: string
+        attributes:
+          description: The name of the attributes to select for the dataset.
```

Comment:
to be clear, these need to be attributes of the primary entity, right? If not, how would I specify a foreign table attribute?

Reply:
Yes, primary entity. added more to the comment.